### PR TITLE
Added support for Clone to rocket::http::uri::Uri

### PR DESCRIPTION
**Pull request by talg**

### CODE REVIEW

File: core/http/src/parse/indexed.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl AsPtr for [u8] {
 }
 
 
-#[derive(PartialEq)]
+#[derive(PartialEq, Clone)]
```
Comment (SergioBenitez) [inline core/http/src/parse/indexed.rs]: This is incorrect as it adds a `Clone` bound to `T`, which is unnecessary. `Indexed` already implements `Clone`; no changes are needed to `indexed.rs`.

File: core/http/src/parse/indexed.rs [hunk 2]
```diff
@@ -175,7 +175,7 @@ impl<'a, T: ?Sized + ToOwned + 'a> Indexed<'a, T>
 }
 
 impl<'a, T: ToOwned + ?Sized + 'a> Clone for Indexed<'a, T> {
-    fn clone(&self) -> Self {
+    default fn clone(&self) -> Self {
```
Comment (SergioBenitez) [inline core/http/src/parse/indexed.rs]: This is unnecessary.

**Comment by talg** [comment]
Good catch, fixed in the latest commit.

**Comment by SergioBenitez** [comment]
Merged in 8172f748b32c79f980020d7dabdff22ee131c46f. Thanks so much! :)

**Pull request closed without merging**